sympy/core/expr.py

```diff
@@ -666,7 +666,7 @@ def check_denominator_zeros(expression):
                     a = expr._random(None, 0, 0, 0, 0)
             except ZeroDivisionError:
                 a = None
-            if a is not None and a is not S.NaN:
+            if a is not None and a not in (S.NaN, S.Infinity, S.NegativeInfinity):
```

Comment:
from sympy.polys.polyutils import illegal
...
if a is not None and a not in illegals:

Reply:
Weirdly enough adding `S.ComplexInfinity`, i.e., using `illegal` (should be in a better location, right? In core somewhere?) leads to that the following test fails:

```
    x = Symbol('x', positive=True)
    e5 = (1 + x)/x
    assert e5.is_constant() is False
```
I realize by checking the blame that I added it a while back, but do not really remember the circumstances. It is stated in the documentation that it is tested for denominators being zero. However, in this case returning `None` should be more correct than `False` since `x` cannot be zero...